From b2c03beed5c6d051c9e3e4aaae7cbec972eda6e3 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Wed, 14 Oct 2026 16:31:52 -0700
Subject: [PATCH] Align Dogecoin subsidy interval and regtest snapshot
 commitments

Preserve the final pinned Doged 100000-block subsidy interval and deterministic regtest snapshots while retaining current snapshot count API. Add the separate optional Avalanche preconsensus gate requested by the coordination adaptation.

diff --git a/src/consensus/params.h b/src/consensus/params.h
index 71056c718058b4d49199f4423c33887533a02727..9cae60a85ba8fd0f57ae948e9d90338562839ee5 100644
--- a/src/consensus/params.h
+++ b/src/consensus/params.h
@@ -73,6 +73,7 @@ struct Params {
      * Unix time used for MTP activation of 15 November 2026 12:00:00 UTC
      * upgrade
      */
+    int64_t avalanchePreconsensusActivationTime = 0x7ffffffe;
     int64_t rttActivationTime = 0x7ffffffe;
     int64_t rttOneBlockActivationTime = 0x7ffffffe;
     int64_t mengerActivationTime;
diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index d74c0e42db77f2fbf008b44f07dc6303e3b3a44f..64714d1185b987808c39ab423122b1bca3e664a4 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -74,7 +74,7 @@ class CMainParams : public CChainParams {
 public:
     explicit CMainParams(const ChainOptions &opts) {
         m_chain_type = ChainType::MAIN;
-        consensus.nSubsidyHalvingInterval = 210000;
+        consensus.nSubsidyHalvingInterval = 100000;
         // 00000000000000ce80a7e057163a4db1d5ad7b20fb6f598c9597b9665c8fb0d4 -
         // April 1, 2012
         consensus.BIP16Height = 0;
@@ -128,7 +128,7 @@ public:
         consensus.gravitonHeight = 0x7ffffffe;
         consensus.phononHeight = 0x7ffffffe;
         consensus.axionHeight = 0x7ffffffe;
-        consensus.wellingtonHeight = 0x7ffffffe;    // keep alive for tests
+        consensus.wellingtonHeight = 0x7ffffffe; // disable BCH transaction rules
         consensus.cowperthwaiteHeight = 900000; // keep alive for tests
         consensus.mengerActivationTime = std::numeric_limits<int64_t>::max();
         consensus.sandersActivationTime = std::numeric_limits<int64_t>::max();
@@ -221,7 +221,7 @@ class CTestNetParams : public CChainParams {
 public:
     explicit CTestNetParams(const ChainOptions &opts) {
         m_chain_type = ChainType::TESTNET;
-        consensus.nSubsidyHalvingInterval = 210000;
+        consensus.nSubsidyHalvingInterval = 100000;
         // 00000000040b4e986385315e14bee30ad876d8b47f748025b26683116d21aa65
         consensus.BIP16Height = 0;
         consensus.BIP34Height = 708658;
@@ -413,6 +413,7 @@ public:
         // Nov 15, 2026 12:00:00 UTC protocol upgrade
         consensus.rttActivationTime = 1731672000;
         consensus.rttOneBlockActivationTime = 1763208000;
+        consensus.avalanchePreconsensusActivationTime = 1763208000;
         consensus.mengerActivationTime = 1794744000;
 
         // May 15, 2027 12:00:00 UTC protocol upgrade
@@ -463,21 +464,21 @@ public:
         m_assumeutxo_data = {
             {.height = 110,
              .hash_serialized =
-                 AssumeutxoHash{uint256S("0xd754ca97ef24c5132f8d2147c19310b7a6b"
-                                         "d136766430304735a73372fe36213")},
+                 AssumeutxoHash{uint256S("0xfcfa07adecbe5f753b9f062b5e5621dcdd9"
+                                         "f998a45968876cb98d350667d745e")},
              .m_chain_tx_count = 111,
              .blockhash =
-                 BlockHash{uint256S("0x47cfb2b77860d250060e78d3248bb05092876545"
-                                    "3cbcbdbc121e3c48b99a376c")}},
+                 BlockHash{uint256S("0xd5a3182b833dca6d0c9bad770890080c2639f077"
+                                    "e0e880c5ab16f1ba3a27b740")}},
             {// For use by test/functional/feature_assumeutxo.py
              .height = 299,
              .hash_serialized =
-                 AssumeutxoHash{uint256S("0xa966794ed5a2f9debaefc7ca48dbc5d5e12"
-                                         "a89ff9fe45bd00ec5732d074580a9")},
+                 AssumeutxoHash{uint256S("0xf4217e2debd5677de2a12a7db17e5f1bbc2"
+                                         "663bea3a28a639300601e135b3cc9")},
              .m_chain_tx_count = 334,
              .blockhash =
-                 BlockHash{uint256S("0x118a7d5473bccce9b314789e14ce426fc65fb09d"
-                                    "feda0131032bb6d86ed2fd0b")}},
+                 BlockHash{uint256S("0x147029b0cd6d9b3847db3660938c8220b24d7db3"
+                                    "069a039f12a07f4322c8d888")}},
         };
 
         chainTxData = ChainTxData{0, 0, 0};
